omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java

```diff
@@ -243,11 +249,41 @@ protected PageableResult doSearch(RequestContext context) {
 				filterByType(orders, context.getType());
 			}
 			
-			return new NeedsPaging<Order>(orders, context);
+			if (StringUtils.isNotBlank(sortParam)) {
+				List<Order> sortedOrder = sortOrdersBasedOnDateActivatedOrDateStopped(orders, sortParam, status);
+				return new NeedsPaging<Order>(sortedOrder, context);
+			}
+			else {
+				return new NeedsPaging<Order>(orders, context);
+			}
 		}
 		
 		return new EmptySearchResult();
 	}
+
+	private static Date getUsableDate(Order order) {
+		return order.getDateStopped() != null ? order.getDateStopped() : order.getAutoExpireDate();
```

Comment:
We should have a final fallback here if dateStopped and getAutoExpireDate are both null... (not sure if that should ever happen, but both fields are allowed to be null in the db so it's *possible* it could happen).  I would just fall back to dateCreated (since it is the one date that is guaranteed never to be null).